workflows/pangenome-generate/pangenome-generate_abpoa.cwl

```diff
@@ -0,0 +1,120 @@
+#!/usr/bin/env cwl-runner
+cwlVersion: v1.1
+class: Workflow
+requirements:
+  ScatterFeatureRequirement: {}
+  StepInputExpressionRequirement: {}
+inputs:
+  seqs: File
+  metadata: File
+  bin_widths:
+    type: int[]
+    default: [ 1, 4, 16, 64, 256, 1000, 4000, 16000]
+    doc: width of each bin in basepairs along the graph vector
+  cells_per_file:
+    type: int
+    default: 100
+    doc: Cells per file on component_segmentation
+  reversed_sort:
+    type: string
+    default: "false"
+outputs:
+  odgiGraph:
+    type: File
+    outputSource: buildGraph/odgiGraph
+  odgiPNG:
+   type: File
+   outputSource: vizGraph/graph_image
+  abpoaGFA:
+    type: File
+    outputSource: induceGraph/abpoaGFA
+#  odgiRDF:
+#    type: File
+#    outputSource: odgi2rdf/rdf
+  readsMergeDedupSortedByQualAndLen:
+    type: File
+    outputSource: dedup_and_sort_by_quality_and_len/sortedReadsFA
+  mergedMetadata:
+    type: File
+    outputSource: dups2metadata/merged
+#  indexed_paths:
+#    type: File
+#    outputSource: index_paths/indexed_paths
+#  colinear_components:
+#    type: Directory
+#    outputSource: segment_components/colinear_components
+steps:
+  dedup_and_sort_by_quality_and_len:
+    in: {readsFA: seqs, reversed_sorting: reversed_sort}
+    out: [sortedReadsFA, dups]
+    run: sort_fasta_by_quality_and_len.cwl
+  induceGraph:
+    in:
+      readsFA: dedup_and_sort_by_quality_and_len/sortedReadsFA
+    out: [abpoaGFA]
+    run: abpoa.cwl
+  buildGraph:
+    in: {inputGFA: induceGraph/abpoaGFA}
+    out: [odgiGraph]
+    run: odgi-build-from-xpoa-gfa.cwl
+  vizGraph:
+    in:
+      sparse_graph_index: buildGraph/odgiGraph
+      width:
+        default: 50000
+      height:
+        default: 500
+      path_per_row:
+        default: true
+      path_height:
+        default: 4
+    out: [graph_image]
+    requirements:
+      ResourceRequirement:
+        ramMin: $(15 * 1024)
+        outdirMin: 10
+    run: ../tools/odgi/odgi_viz.cwl
+  # odgi2rdf:
+  #  in: {odgi: buildGraph/odgiGraph}
+  #  out: [rdf]
+  #  run: odgi_to_rdf.cwl
```

Comment:
was this commented out just for debugging purposes, or do we no longer need it?

Reply:
Thanks for taking a look @mr-c! This part was commented out because it used spodgi, which works by loading the entire input into memory, leading to scalability issues. There is an alternative (https://github.com/AndreaGuarracino/rs-spodgi), and I've already prepared a Dockerfile for that, but it has not yet been put into production on PubSeq.